.github/workflows/deploy_pages.yml

```diff
@@ -0,0 +1,215 @@
+               owner: context.repo.owner,
+               repo: context.repo.repo,
+               run_id: ${{github.event.workflow_run.id }},
+            });
+            const reports = artifacts.data.artifacts.filter((artifact) => {
+              return artifact.name.startsWith('vrt')
+            });
+
+            if (reports) {
+              const directory = path.join('${{github.workspace}}', 'vrt');
+              fs.mkdirSync(directory);
+
+              for (const artifact of reports) {
+                const download = await github.actions.downloadArtifact({
+                  owner: context.repo.owner,
+                  repo: context.repo.repo,
+                  artifact_id: artifact.id,
+                  archive_format: 'zip',
+                });
+                fs.writeFileSync(path.join(directory, `${artifact.name}.zip`), Buffer.from(download.data));
+              }
+            }
+
+            return reports.length > 0;
+      - run: |
+          unzip vrt/*.zip
+          rm vrt/*.zip
+          cp -R vrt/* github-pages
+        if: ${{ steps.vrt.outputs.result == true }}
+      - name: Download axe reports
+        uses: actions/github-script@v6
+        id: storybook
+        with:
+          script: |
+            const fs = require('fs');
+
+            const artifacts = await github.actions.listWorkflowRunArtifacts({
+               owner: context.repo.owner,
+               repo: context.repo.repo,
+               run_id: ${{github.event.workflow_run.id }},
+            });
+            const artifact = artifacts.data.artifacts.find((artifact) => {
+              return artifact.name == 'aat';
+            });
+
+            if (artifact) {
+              const download = await github.actions.downloadArtifact({
+                owner: context.repo.owner,
+                repo: context.repo.repo,
+                artifact_id: artifact.id,
+                archive_format: 'zip',
+              });
+              fs.writeFileSync('${{github.workspace}}/aat.zip', Buffer.from(download.data));
+            }
+
+            return !!artifact;
+      - run: |
+          unzip aat.zip -d aat-artifact
+          mkdir github-pages/aat
+          cp -R aat-artifact/ github-pages/aat/
+      - name: Check if any artifacts exist
+        if: >
+          steps.storybook.outputs.result != true &&
+          steps.docs.outputs.result != true &&
+          steps.vrt.outputs.result != true &&
+          steps.aat.outputs.result != true
+        run: exit 1
+      - name: Upload artifact
+        uses: actions/upload-artifact@main
+        with:
+          name: github-pages
+          path: artifact.tar
+
+  deploy:
+    if: ${{ github.repository == 'primer/react' }}
+    runs-on: ubuntu-latest
+    needs: upload
+    environment:
+      name: github-pages
+      url: ${{ steps.deployment.outputs.page_url }}
+    permissions:
+      contents: read
+      pages: write
+      id-token: write
+    outputs:
```

Comment:
How would the output look with these changes? Would it go to the main page of the docs as we have now?

Since you and @langermank brought the Storybook view deployment link in, I can't help but think accessing all these artefacts' links (Axe and VRT results) on the PR page 😋

Reply:
@broccolinisoup should be the same as it is currently! (Or at least that's my intent haha) 🤞